packages/serverpod_client/lib/src/serverpod_client_shared_private.dart

```diff
@@ -30,6 +32,17 @@ dynamic getExceptionFrom({
   required SerializationManager serializationManager,
   required int statusCode,
 }) {
+  if (data.isEmpty) {
```

Comment:
Question: Shouldn't this also apply if we fail to decode the passed object?

Basically in both cases below?

Reply:
Suggestion: add some unit tests for this code that validate this works as expected. Then we could test the case when we fail to decode for example.